### DISCUSSION TITLE
oo/abstract-vs-interface

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#92] [Announcements]
# oo/abstract-vs-interface

Java程序员进阶之路，小白的零基础Java教程，认真聊聊 Java抽象类和接口的区别

https://tobebetterjavaer.com/oo/abstract-vs-interface.html

**Comment by deqiangm** [comment]
>1、抽象类可以有方法体的方法，但接口没有。

这个准确吗？我记得你上一篇说过，java8之后抽象类可以有default的方法，有方法体

**Reply by itwanger** [reply]
（Java 8 以前）

**Reply by itwanger** [reply]
需要加上这个

**Comment by ismyxh** [comment]
> 抽象类可以提供成员方法的实现细节，而接口中只能存在public abstract 方法；

这一句也可以加上（Java 8 以前）。

还有就是：既然接口可以有方法体，那从设计层面上说，他们的区别还是这样的吗
